src/url.cpp

```diff
@@ -298,4 +305,76 @@ namespace Url
         return *this;
     }
 
+    Url& Url::relative(const std::string& other)
```

Comment:
The sense of this is the other way (though perhaps we should change that). For example,

```c++
Url base("http://base.com/")
base.relative("/some/relative/path");
// Now base is "http://base.com/some/relative/path"
```

This isn't particularly amenable to re-use, but this is how it's written in `url-py`. Instead, perhaps we should do this:

```c++
Url relative("/some/relative/path");
relative.relative("http://base.com/");

// Alternatively, in the vein of your overload, now we can reuse base
Url relative("/some/relative/path");
Url base("http://base.com/");
relative.relative(base);
```